app/channel.go

```diff
@@ -596,10 +596,48 @@ func (a *App) postChannelPrivacyMessage(user *model.User, channel *model.Channel
 	return nil
 }
 
-func (a *App) RestoreChannel(channel *model.Channel) (*model.Channel, *model.AppError) {
+func (a *App) RestoreChannel(channel *model.Channel, userId string) (*model.Channel, *model.AppError) {
+	var user *model.User
+	if userId != "" {
+		var err *model.AppError
+		user, err = a.Srv.Store.User().Get(userId)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if channel.DeleteAt == 0 {
+		err := model.NewAppError("undeleteChannel", "api.channel.undelete_channel.undeleted.app_error", nil, "", http.StatusBadRequest)
+		return nil, err
+	}
+
+	if user != nil {
+		T := utils.GetUserTranslations(user.Locale)
+
+		post := &model.Post{
+			ChannelId: channel.Id,
+			Message:   fmt.Sprintf(T("api.channel.undelete_channel.unarchived"), user.Username),
+			Type:      model.POST_CHANNEL_UNDELETED,
+			UserId:    userId,
+			Props: model.StringInterface{
+				"username": user.Username,
```

Comment:
Why do we need to include the `username`?

Reply:
i think we need it because the post says "{username} unarchived the channel"?

Reply:
You are building the message in https://github.com/mattermost/mattermost-server/pull/13349/files#diff-243b54bc23ef263e3d67c7f01d3972a0R621 with the user name. Why do you need it in `Props`?

Reply:
youre right. we are grabbing it from the user object